### PR TITLE
add datasource to compute the list of fields in a collection

### PR DESCRIPTION
**Pull request by benoittoulme**

### CODE REVIEW

File: provider/resource_collection.go [hunk 1]
```diff
@@ -1103,3 +1137,73 @@ func populateCollectionEntriesResourceData(ctx context.Context, c *models.Collec
...
+
+	for _, item := range arr {
+		item_, ok := item.(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("Expected map in array body return type!")
+		}
+		row := []map[string]interface{}{}
+		for k, _ := range item_ {
+			// Perhaps do not include internal fields...
+			if k[0] != '_' || k == "_idx" || (k == "_key") {
```
Comment (masha1997) [inline provider/resource_collection.go:1183]: Why just k[0] != '_'  don't cover this purposes?

Comment (masha1997) [inline provider/resource_collection.go:1183]: Why just k[0] != '_'  don't cover this purposes?

File: provider/resource_collection.go [hunk 2]
```diff
@@ -858,7 +858,7 @@ func populateCollectionEntryResourceData(ctx context.Context, c *models.Collecti
 	}
 	dataRes, ok := obj.(map[string]interface{})
 	if !ok {
-		return fmt.Errorf("Expected map body return type!")
+		return fmt.Errorf("expected map body return type")
```
Comment (benoittoulme) [inline provider/resource_collection.go:861]: image

**Review by dmitrykruglov — APPROVED** [review]
Looks good

**Pull request merged**